pkg/util/must/must.go

```diff
@@ -176,6 +182,19 @@ var (
 	OnFail func(ctx context.Context, err error)
 )
 
+// DisableFatalAssertions disables fatal assertions for the duration of a test,
+// re-enabling them at the end. This is useful for tests that check assertion
+// failures as errors.
+func DisableFatalAssertions(t testing.TB) {
```

Comment:
This adds a testing.TB dependency to non-testing code.
Maybe have a small local interface instead?

Reply:
Yeah, we already do that in a bunch of places (see e.g. `storage.DisableMetamorphicSimpleValueEncoding`) so I just yoloed it, but let's do better.

Reply:
Looks like I'm to blame for `DisableMetamorphicSimpleValueEncoding ` too, via #81284, must've missed that it wasn't a test file. :( I'll clean it up tomorrow.